packages/bsky/src/api/app/bsky/graph/getList.ts

```diff
@@ -68,9 +69,41 @@ const hydration = async (
       params.hydrateCtx,
     ),
   ])
+
+  const list = listState.lists?.get(listUri)
+  const ownerDid = new AtUri(listUri).hostname
+  if (
+    list?.record.purpose === 'app.bsky.graph.defs#referencelist' &&
+    // We show all users regardless of blocks if the viewer is the owner of the list, so no need to hydrate them
+    params.hydrateCtx.viewer !== ownerDid
+  ) {
+    const pairs = new Map()
+    for (const { did } of listitems) {
+      pairs.set(did, [did, ownerDid])
+    }
+    const bidirectionalBlocks =
+      await ctx.hydrator.hydrateBidirectionalBlocks(pairs)
+    return mergeStates(listState, { ...profileState, bidirectionalBlocks })
```

Comment:
If `profileState` contains some `bidirectionalBlocks` they would get clobbered here, so we'd be safer with something like `mergeManyStates(listState, profileState, { bidirectionalBlocks })`.